### PR TITLE
Bug/cyclic super

### PR DESCRIPTION
**Pull request by JGets**
Fixes an issue where recursive / cyclic calls to a (super) method do not actually invoke cyclically as expected.

### CODE REVIEW

File: JS.js [hunk 1]
```diff
@@ -630,6 +635,37 @@ JS.class.initialFieldValue = function(instance, field) {
...
 
 /**
+ * Makes a proxy function that, if nessecary, updates the method's `superStack` with the current
+ * invocation of the method to allow for recursive/cyclic super calls
+ * @param  {function} method
+ * @return {function}
+ */
+function makeSuperStackUpdaterProxy(method) {
+	return function proxy() {
+		var updatedStack = false;
```
Comment (ColinRhodes) [inline JS.js]: var => let

File: JS.js [hunk 2]
```diff
@@ -630,6 +635,37 @@ JS.class.initialFieldValue = function(instance, field) {
 };
 
 /**
+ * Makes a proxy function that, if nessecary, updates the method's `superStack` with the current
```
Comment (ColinRhodes) [inline JS.js]: Needs better description of how the superStack and instanceStack relate

File: JS.js (hunk 1 above)
Comment (JGets) [inline JS.js]: Converted `makeStackUpdaterProxy` to use strict mode in 18a85e91

File: JS.js (hunk 2 above)
Comment (JGets) [inline JS.js]: Updated comments in 18a85e9

**Pull request merged**